### ISSUE TITLE
Descriptor type validation doesn't handle OpGroupDecorate

### ISSUE THREAD

**Issue by phaulos** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1549]
`dEQP-VK.spirv_assembly.instruction.compute.decoration_group.all` is failing with:

```
ERROR: Type mismatch on descriptor slot 0.0 (used as type `ptr to uniform struct of (oddtype)`) but descriptor of type VK_DESCRIPTOR_TYPE_STORAGE_BUFFER (code 0x0000000d from SC at Unknown:2631)
...
```

Looking at the SPIR-V `BufferBlock` decoration is applied to `0.0` through `OpGroupDecorate`. This is not something glslang does so it might be that validation simply doesn't handle it.

**Comment by chrisforbes** [comment]
Indeed, SC has no idea about group decorations at all. This is going to take a bit of work.

**Comment by chrisforbes** [comment]
Working out with spirv-tools side how best to deal with this.

**Comment by gwihlidal** [comment]
Any update on this? We're hitting this error in many of our shaders.

**Comment by mark-lunarg** [comment]
@chrisforbes, any update? Still applicable?
